server/src/main/java/com/example/service/UploadLogService.java

```
@@ -30,19 +35,22 @@
 		return uploadLogs;
 	}
 
-	public UploadLog getUploadLogById(String id) {
-		return uploadLogRepository.findById(id)
-		                          .orElse(null);
+	public UploadLogDto getUploadLogById(String id) {
+		UploadLog uploadLog = uploadLogRepository.findById(id)
```

we can upgrade the repository that throw exceptions in case of missing return (like we have done for profileRepo and machineRepo)

---

not sure what you mean